pkg/cmd/marathon/app/app_add.go

```diff
@@ -1,18 +1,52 @@
 package app
 
 import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/dcos/dcos-core-cli/pkg/marathon"
+
 	"github.com/dcos/dcos-cli/api"
-	"github.com/dcos/dcos-core-cli/pkg/cmd/marathon/python"
 	"github.com/spf13/cobra"
 )
 
 func newCmdMarathonAppAdd(ctx api.Context) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "add",
 		Short: "Add an application.",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return python.InvokePythonCLI(ctx)
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			client, err := marathon.NewClient(ctx)
+			if err != nil {
+				fmt.Fprintf(ctx.ErrOut(), "%s", err)
+				return
+			}
+			if len(args) > 0 {
+				err = marathonAppAdd(ctx, *client, args[0])
+			} else {
+				err = marathonAppAdd(ctx, *client, "")
+			}
+			if err != nil {
+				os.Exit(1)
```

Comment:
Exithing without any error message? This might be confusing. Why not using `RunE` like it's done in other commands?

Reply:
```
func newCmdMarathonAppAdd(ctx api.Context) *cobra.Command {
	cmd := &cobra.Command{
		Use:   "add",
		Short: "Add an application.",
		Args:  cobra.MaximumNArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			client, err := marathon.NewClient(ctx)
			if err != nil {
				return err
			}
			if len(args) > 0 {
				err = marathonAppAdd(ctx, *client, args[0])
			} else {
				err = marathonAppAdd(ctx, *client, "")
			}
			return err
		},
	}
	return cmd
}

func marathonAppAdd(ctx api.Context, client marathon.Client, appFile string) error {
	app, err := client.AddApp(ctx, appFile)
	if err != nil {
		if err == marathon.ErrCannotReadAppDefinition {
			return fmt.Errorf("Can't read from resource: %s.\nPlease check that it exists.\n", appFile)
		}
		if syntaxError, ok := err.(*json.SyntaxError); ok {
			return fmt.Errorf("Error loading JSON: %s\n", syntaxError.Error())
		}
		return fmt.Errorf("%s\n", err.Error())
	}
	fmt.Fprintf(ctx.Out(), "Created deployment %s\n", app.Deployments[0]["id"])
	return nil
}
```

Reply:
Oh shit, that was a copy'n'paste mistake.

Reply:
After moving to `RunE` you can easily add tests

Reply:
I don't agree on giving up tight controler over the messages a user sees. Also, using just `error`s here will force us to format user-visible output according to common error-formatting rules such as not starting an error message with capital letters or ending it with a `.` which doesn't make sense in this scenario.